%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%Preamble
\documentclass{article}

%Dependencies
\usepackage[left]{lineno}
\usepackage{titlesec}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{color,soul}
\usepackage{ogonek}
\usepackage{float}


% Packages from Am. Nat. Template
%\usepackage{times}
\usepackage[sc]{mathpazo} %Like Palatino with extensive math support
\usepackage{fullpage}
\usepackage[authoryear,sectionbib,sort]{natbib}
\linespread{1.7}
\usepackage{lineno}

% Graphics
\usepackage{graphicx}
\graphicspath{{../output/figures/}.pdf}

% New commands: fonts
%\newcommand{\code}{\fontfamily{pcr}\selectfont}
%\newcommand*\chem[1]{\ensuremath{\mathrm{#1}}}
\newcommand\numberthis{\addtocounter{equation}{1}\tag{\theequation}}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Title Page

\title{Consequences of genetic linkage for the maintenance of sexually anatagonistic polymorphism in hermaphrodites}
\author{Colin Olito}
\date{\today}

\begin{document}
\maketitle


\noindent{} Centre for Geometric Biology, School of Biological Sciences, Monash University, Victoria 3800, Australia.
\noindent{} Corresponding author e-mail: colin.olito@gmail.com

\bigskip

\noindent{} \textit{Running Head}: Linkage and SA polymorphism in hermaphrodites

\bigskip

\noindent{} \textit{Keywords}: Balancing selection, genetic linkage, intralocus sexual conflict, recombination, sexual dimorphism, two-locus model

\bigskip

\noindent{} \textit{Manuscript type}: Brief Communication

\bigskip


% Set line number options
\linenumbers
\modulolinenumbers[1]
\renewcommand\linenumberfont{\normalfont\small}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Main Text

\newpage{}
\section*{Abstract (200 word limit)}

\noindent{} ...

\newpage{}


\section*{Introduction}

Due to fundamental differences in their reproductive biology arising from anisogamy, males and females rarely share the same optimal phenotype (\citealt{XXX}). When phenotypic optima differ between males and females, pleiotropic effects among genes expressed by both sexes can result in sexually antagonistic selection (SA hereafter), where beneficial alleles for one sex have deleterious fitness effects in the other (\citealt{Kidwell1977, Rice1992, ConnClark2012}). SA alleles have been shown to play an important role in the maintenance of genetic variation in fitness, the evolution of reproductive and life-history traits, and genome evolution in a variety of theoretical and empirical contexts (\citealt{Barson2015, BondChen2009, CoxCals2010, CharlesworthHughes1999, ConnClark2012, Fry2010, Prout2000, Rice1992, RiceChipp2001}). 

The existence of physically separate sexes makes the relevance of SA selection to dioecious species intuitive, and this is reflected in the traditional emphasis on SA polymorphism in this group (e.g.~\citealt{Kidwell1977, Rice1992, Prout2000, ConnClark2012}). However, alleles with opposing fitness effects through male and female sex functions represent analogous genetic constraints on fitness for hermaphrodites (\citealt{Abbott2011, JordanConn2014,Tazzyman2015}). For hermaphroditic individuals, both the male and female sex functions must be accommodated by a single phenotype, and there is ample scope for traits with a shared genetic basis to constrain fitness through each sex function (\citealt{Abbott2011, Barrett2002, Conner2006, Sicard2011}). Furthermore, fitness trade-offs between sex functions, and thus SA alleles, can have diverse consequences for the evolution of reproductive traits (e.g.~floral and inflorescence morphology) as well as sexual and mating systems in hermaphrodites (\citealt{LloydWebb1986,WebbLloyd1986, Barrett2002, Abbott2011, Charlesworth1978, HarderBarrett2006, Goodwillie2005}).

The fate of SA alleles in hermaphrodites is complicated by the fact that individuals may pass on genes to subsequent generations through a combination of outcrossing and self-fertilization (\citealt{Goodwillie2005, JarneAuld2006, Igic2005, JordanConn2014}). Self-fertilization has several important theoretical consequences for the maintenance of SA genetic variation (\citealt{JordanConn2014, KimuraOhta1971}). Selfing diminishes the opportunity for balancing selection to maintain SA polymorphism, and thereby reduces the sensitivity of SA polymorphism to dominance (\citealt{JordanConn2014, Tazzyman2015}). Selfing also creates a ``female-bias'' in the net direction of selection because there is reduced opportunity for selection to act via the male sex function (\citealt{Charlesworth1978, JordanConn2014}; but see \citealt{Tazzyman2015} for alternative model assumptions). This skews the parameter space where SA polymorphism can be maintained towards female-beneficial alleles (\citealt{JordanConn2014}). 

Self-fertilization carries with it a variety of other genomic consequences for hermaphrodites, including a reduction in the effective rate of recombination among neighboring loci (reviewed in \citealt{Wright2008}). The effect of selfing on effective recombination is of particular interest because this should drive selection on linked allelic combinations, potentially preserving polymorphisms that would otherwise experience purifying selection at a single locus (\citealt{CITE}). For dioecious species, tight linkage between SA loci is predicted to expand the parameter space where SA polymorphism is maintained (\citealt{Patten2010}). Thus, for hermaphrodites there is the potential for reduced recombination to compensate for the loss in SA polymorphism due to increased selfing. However, the conditions under which each of these two countervailing effects of self-fertilization will be more influential in maintaining SA polymorphism have yet to be explored. 

Here I use a two-locus model of SA selection with partial selfing to investigate the joint influence of linkage, self-fertilization, and dominance on the maintainance of SA polymorphism in hermaphroditic species. In the two-locus context, reduced effective recombination due to selfing significantly increases the parameter space where balancing selection is predicted to maintain SA polymorphism relative to single-locus models. In particular, linkage expands the conditions where male-beneficial alleles are able to invade, partially compensating for the ``female-bias'' in selection introduced by selfing.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Model}

Consider a genetic system involving two diallelic autosomal loci $\mathbf{A}$ and $\mathbf{B}$, with alleles $A$ and $a$, $B$ and $b$, that recombine at a rate $r$ in a large population of simultaneous hermaphrodites. The rate of self-fertilization ($C$) in the population is independent of the genotype at the loci in question (a `fixed' selfing model \textit{sensu} \citealt{Holden1979,CaballeroHill1992, JordanConn2014}). Generations are assumed to be discrete and non-overlapping, with selection occuring on diploid adults before mating. Let $x_i$ and $y_i$ denote the frequencies of the four possible haplotypes $[AB, Ab, aB, ab]$ among male and female gametes respectively. Both loci are under sexually antagonistic selection such that $A$ and $B$ represent male-beneficial alleles, while $a$ and $b$ represent female-beneficial alleles (\citealt{Kidwell1977}). The fitness of offspring formed by the union of the $i$th female and $j$th male gametic haplotypes, $w_{k,ij}$ (where $k \in [m,f]$), is assumed to equal the product of the fitnesses at $\mathbf{A}$ and $\mathbf{B}$ (Table~\ref{Table:Fitness}; this parameterization follows \citealt{Patten2010}). Following convention for SA models, sex-specific selection coefficients are constrained to be $0 < s_k < 1$ (e.g.~\citealt{Kidwell1977}).

The evolutionary trajectory of genotype frequencies in this scenario can be described by a system of ten recursion equations (\citealt{Holden1979, JordanConn2014}). However, it is possible to approximate the evolutionary trajectories of haplotypes in this system under weak selection (See Appendix A in the Supplementary Information). For partially selfing populations under weak selection, the rate of allele frequency change due to selection should be slow relative to the rate at which genotypes approach equilibirium under non-random mating (\citealt{Nagylaki1997}). Under this assumption, it may be appropriate to use a separation of timescales (\citealt{OttoDay2007}), and calculate the quasi-equilibrium (QE) genotypic frequencies in the absence of selection. The genotypic recursions for allele frequency change across generations can then be approximated by substituting into them the QE frequencies, yielding a reduced system of four haplotype recursions. 

Using this approach, I model the evolution of the four-haplotype system $q_i = [AA, Ab, aB, ab]$, where the QE adult genotypic frequencies are denoted by $\phi_{ij}$. The recursions giving the haplotype frequencies in the next generation are then

\begin{align*} \label{eq:QEhapRec}
	q'_1 &\approx (1 - C) \frac{(x_1 + y_1)}{2} + C \bigg( \frac{x_1 - r(\phi{14} - \phi{23})}{2 \overline{w}_f} \bigg) \\
	q'_2 &\approx (1 - C) \frac{(x_2 + y_2)}{2} + C \bigg( \frac{x_2 + r(\phi{14} - \phi{23})}{2 \overline{w}_f} \bigg) \\
	q'_3 &\approx (1 - C) \frac{(x_3 + y_3)}{2} + C \bigg( \frac{x_3 + r(\phi{14} - \phi{23})}{2 \overline{w}_f} \bigg) \\
	q'_4 &\approx (1 - C) \frac{(x_4 + y_4)}{2} + C \bigg( \frac{x_4 - r(\phi{14} - \phi{23})}{2 \overline{w}_f} \bigg), \numberthis
\end{align*}

\noindent{} where $x_i$ and $y_i$ are functions $f(C, s_k, h_k, \phi_{ij})$ describing the haplotype frequencies in male and female gametes, and $\overline{w}_f$ is the population average fitness through female function (see Appendix A in the the Supplementary Information for a full development of the recursions). To examine how well the system of QE haplotype recursions approximate changes in the genotypic frequencies, I also performed determininstic simulations using the full system of genotypic recursions (the full set of exact recursions can be found in Appendix A in the the Supplementary Information). The QE haplotype recursions approximated the evolutionary trajectories of genotypic frequencies remarkably well, even under strong selection (Figs.~\ref{fig:addSim} and \ref{fig:domRevSim} in the Supporting Information); I therefore focus the analysis on the analytic QE results. 

To identify the parameter conditions under which balancing selection is predicted to maintain SA polymorphism, I evaluate the stability of the system of haplotype recursions Eq(\ref{eq:QEhapRec}) for populations initially fixed for male-beneficial or female-beneficial alleles (i.e.~stability was assessed at the boundary haplotype frequences $[AB]=1$ and $[ab]=1$). For these boundary equilibria, the leading eigenvalue of the system of recursions, $\lambda_L$, describes the rate of change of the frequencies of rare haplotypes, and therefore whether new mutations will be able to invade (\citealt{OttoDay2007}). Balancing selection is predicted to maintain SA polymorphism when $\lambda_L > 1$ for both boundary equilibria (\citealt{Prout1968}). Two candidate leading eigenvalues emerge from the analysis. The first describes invasion at a single locus, and is a function of the sex-specific selection parameters ($s_k$, $h_k$) and the selfing rate ($C$). Balancing selection at both loci occurs whenever single-locus criteria for balancing selection are met because the selection parameters are identical for both loci (if invasion can occur at one locus, it can also occur at the other). If the two SA loci are physically linked, the conditions for balancing selection are expanded beyond the single-locus expectation. This condition is described by the second candidate leading eigenvalue, which involves the recombination rate, $r$, in addition to $s_k$, $h_k$, and $C$.

I focus the analysis on representative, and biologically plausible, dominance scenarios that have been recently explored in the single-locus context (\citealt{Kidwell1977, Fry2010, Prout2000, JordanConn2014}). These correspond to: (1) additive fitness effects ($h_f = h_m = 1/2$), as is commonly observed for small to intermediate effect alleles (\citealt{AgrawalWhitlock2011}); and (2) partially recessive fitness effects yielding a ``dominance reversal'' ($h_f = h_m = 1/4$), which are commonly predicted by a variety of fitness landscape models (\citealt{Manna2011, ConnClark2014}), and are predicted to evolve under some conditions for SA alleles (\citealt{Spencer2016}).


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Results}


Consideration of invasion at a single locus provides a useful baseline for comparison with the two-locus case. Analysis of the first candidate leading eigenvalue yields the general invasion criteria at a single locus, where balancing selection requires that 

\begin{equation} \label{eq:genInv}
	\frac{s_m(C - 1)(2 h_m(C - 1) - C)}{s_m(C - 1)(2h_m(C - 1) - C) + (C + 1)(2 - C + 2h_f(C - 1))} < s_f < \frac{s_m(C - 1)(2 - C + 2 h_m(1 - C))}{(C + 1)(2 h_f(C - 1) - C)(s_m - 1)}.
\end{equation}

\noindent{} Under additive fitness effects ($h_f=h_m=1/2$) and obligate outcrossing ($C=0$), Eq(\ref{eq:genInv}) reduces to the invasion criteria from previous single-locus models (\citealt{JordanConn2014, Tazzyman2015, Patten2010, Kidwell1977}). The corresponding funnel-shaped parameter space where balancing selection at a single locus is predicted to maintain SA polymorphism is shown in Fig.~\ref{fig:funnelPlots} by the shaded regions bounded by black solid lines. 

Increased self-fertilization broadens the scope for linkage to maintain SA polymorphism that would otherwise be lost in a single-locus model. Invasion criteria involving both loci are more complex than the single-locus case. As in \citet{Patten2010}, under perfect linkage ($r=0$), the two-locus invasion criteria expands the parameter space conducive to balancing selection, describing a funnel-shaped region that subsumes the region described by the single-locus criteria. With weaker linkage ($r > 0$), stronger selection is required for the two-locus invasion criteria to expand the parameter space conducive to balancing selection (Fig.~\ref{fig:funnelPlots}, greyscale lines). This is most restrictive under obligate outcrossing, where linkage must be tight ($r < 0.2$ for additive effects, $r < 0.1$ for dominance reversal), for the two-locus invasion criteria to increase SA polymorphism beyond the single-locus case (Fig.~\ref{fig:funnelPlots}A,D; \citealt{Patten2010}). If selection on individual SA loci is weak ($s_f,s_m < 0.1$), extremely tight linkage is required to significantly expand the parameter space where balancing selection is predicted beyond that of single-locus models (Figs.~\ref{fig:wkFunnelPlots}, \ref{fig:wkPolymorhism} in the Supplementary Information). However, the magnitude of the increase is still substantial under tight linkage, particularly for additive fitness effects.

Increased selfing has two important consequences in the two-locus context. As predicted for the single-locus case, selfing biases the net direction of selection towards female interests, reducing the opportunity for male-beneficial alleles to invade at a single locus (Fig.~\ref{fig:funnelPlots}B,C,E,F; \citealt{Charlesworth1978, JordanConn2014}; but see \citealt{Tazzyman2015}). However, the concomitant decrease in effective recombination among double heterozygotes partially compensates for this ``female-bias'' in selection in several ways. First, linkage increases the effective strength of selection at both loci, which is more permissive of balancing selection (\citealt{Patten2010}). Second, reduced recombination slows the break-up of haplotypes pairing male-beneficial with male-beneficial (and female-beneficial with female-beneficial) alleles at each locus, and increases the likelihood that male-beneficial alleles at each locus are paternally inherited (and vice versa for female-beneficial alleles) (\citealt{Patten2010, Ubeda2010}). With higher selfing, this effectively shelters male-beneficial alleles from increasingly strong selection through the female sex function. The net effect is to expand the parameter space where balancing selection is predicted to maintain SA polymorphism, particularly for male-beneficial alleles (Fig.~\ref{fig:funnelPlots}, upper greyscale lines). The effect of reduced recombination is strongest at higher selfing rates ($C > 0.5$), where interaction between SA loci effectively expands the parameter space where male-beneficial alleles can invade beyond the single-locus case, even under free recombination (Fig.~\ref{fig:funnelPlots}B-C,E-F, Fig.~\ref{fig:polymorphism}B,D). 

Under additive fitness effects, the increase in SA polymorphism due to linkage can offset the loss in parameter space attributable to invasion at a single locus up to a selfing rate of about $C = 0.5$ (Fig.~\ref{fig:polymorphism}A). At higher selfing rates ($C > 0.5$), the total parameter space where balancing selection can occur declines, but the relative increase in SA polymorphism predicted by the two-locus model relative to the single-locus case becomes increasingly pronounced, even under free recombination (Fig.~\ref{fig:polymorphism}A,B). Under dominance reversal conditions ($h_f,h_m < 1/2$) the sex-specific fitness costs of SA alleles are partially masked, and balancing selection can maintain SA polymorphism over a broader range of parameter conditions due to net overdominance in fitness (\citealt{ConnClark2012, ConnClark2014}). As a consequence, the effect of linkage is somewhat dampened, and increased selfing always results in a decrease in the total parameter space where SA polymorphism is predicted (Fig.~\ref{fig:polymorphism}C). However, the relative increase in the two-locus relative to the single-locus case remains for higher selfing rates ($C > 0.5$), even under free recombination (Fig.~\ref{fig:polymorphism}D).



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Discussion}


Accounting for linkage between SA loci yields several theoretical insights regarding the maintenance of SA polymorphism in hermaphroditic organisms. As in the dioecious case (\citealt{Patten2010}), including a second SA locus increases the scope for SA polymorphism beyond the predictions from single-locus models by two main mechanisms; (1) increased effective strength of selection at each locus, and (2) stable linkage disequilibrium arising from the admixture of gamete pools experiencing sexually antagonistic selection. In the context of hermaphroditic populations, both mechanisms are influenced by the reduction in effective recombination rate introduced by self-fertilization (\citealt{Wright2008}), becoming increasingly permissive of SA polymorphism at higher selfing rates.

Provided that either linkage is tight or selection is strong, the reduction in the effective recombination rate due to selfing significantly expands the parameter space where balancing selection is predicted to maintain SA polymorphism beyond that of single-locus models (doubling the parameter space under some conditions). The dominance of sex-specific fitness costs of SA alleles alters the relative importance of recombination and net overdominance as proximal mechanisms underlying balancing selection at multiple SA loci, but does not dramatically influence the increase in SA polymorphism relative to single-locus models. Thus, although balancing selection at SA loci is least likely in highly selfing species, it may still provide a plausible mechanism, along with recurrent mutation (\citealt{JordanConn2014}), for the maintenance of genetic variation at linked SA loci in populations with intermediate to high selfing rates. A corrollary of this prediction is that SA genes may be more tightly clustered on the genomes of partially selfing species relative to outcrossing ones because linkage increasingly promotes polymorphism in partial selfers (\citealt{Patten2010}).

As the frequency of self-fertilization increases, linkage among SA loci primarily compensates for the increasing ``female-bias'' in selection, and the associated decrease in SA polymorphism predicted by single-locus models (\citealt{Charlesworth1978, JordanConn2014}), by expanding the parameter space where male-beneficial alleles can invade (but again see \citealt{Tazzyman2015} for alternative assumptions regarding selection on selfed gametes). In this way, linkage among SA loci may have bearing on the prevalence of mixed mating systems, which are represented by a large fraction of hermaphroditic plants and animals (\citealt{JarneAuld2006, Goodwillie2005, Igic2005}). Various theoretical explanations have been proposed for mixed mating, including (but not limited to) reproductive assurance (\citealt{Lloyd1979}), purging of deleterious alleles and reduced inbreeding depression (\citealt{SchemskeLande1985}), the timing of inbreeding depression in the life cycle (\citealt{HarderRoutely}), reproductive compensation (\citealt{HarderRoutely, PorcherLande2005}), and frequency dependence (\citealt{Holsinger1991}) (reviewed in \citealt{Goodwillie2005, HarderBarrett2006}). While the female-bias in selection introduced by selfing is consistent with the evolution of ``selfing syndromes'' (\citealt{Sicard2011}), the increased scope for male-beneficial SA alleles to invade provides another possible explanation for the persistence of traits promoting male gamete dispersal and performance in partially selfing populations (\citealt{Barrett2002, Goodwillie2005, HarderBarrett2006}). Given that change in SA allele frequencies is often predicted to be slow and dominated by drift (\citealt{ConnClark2011, ConnClark2012, ConnClark2014}), the loss of male-beneficial alleles may also be slow and stochastic in partially selfing species with linked SA loci, regardless of whether mixed mating is an evolutionary stable strategy.







%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgments}
CO thanks C.~Venables, J.~Cally, and T.~Connallon for helpful discussions and feedback. The idea for this article was conceived during discussion among Drs.~Tim Connallon, Crispin Jordan, and Jessica K.~Abbott, and CO gratefully acknowledges their encouragement to develop it. This work was funded by a Monash University Dean's International Postgraduate Student Scholarship to CO, and by the Monash University Centre for Geometric Biology.

\newpage{}





%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Tables}
\renewcommand{\thetable}{\arabic{table}}
\setcounter{table}{0}

\begin{table}[h]
\caption{Two-locus fitness expressions for adult females ($w_{f,ij}$) and males ($w_{m,ij}$). Rows and columns indicate the haplotype inherited from mothers and fathers respectively.}
\label{Table:Fitness}
\centering
\begin{tabular}{l c c c c} \hline
Haplotype & $y_1 = AB$ & $y_2 = Ab$ & $y_3 = aB$ & $y_4 = ab$ \\
\hline
Females & & & & \\
$x_1 = AB$ & $(1-s_f)^2$ & $(1 - s_f)(1 - h_f s_f)$ & $(1 - s_f)(1 - h_f s_f)$ & $(1 - h_f s_f)^2$ \\
$x_2 = Ab$ & $(1 - s_f)(1 - h_f s_f)$ & $(1-s_f)$ & $(1 - h_f s_f)^2$ & $(1 - h_f s_f)$ \\
$x_3 = aB$ & $(1 - s_f)(1 - h_f s_f)$ & $(1 - h_f s_f)^2$ & $(1-s_f)$ & $(1 - h_f s_f)$ \\
$x_4 = ab$ & $(1 - h_f s_f)^2$ & $(1 - h_f s_f)$ & $(1 - h_f s_f)$ & $1$ \\
Males & & & & \\
$x_1 = AB$ & $1$ & $(1 - h_m s_m)$ & $(1 - h_m s_m)$ & $(1 - h_m s_m)^2$ \\
$x_2 = Ab$ & $(1 - h_m s_m)$ & $(1-s_m)$ & $(1 - h_m s_m)^2$ & $(1 - s_m)(1 - h_m s_m)$ \\
$x_3 = aB$ & $(1 - h_m s_m)$ & $(1 - h_m s_m)^2$ & $(1-s_m)$ & $(1 - s_m)(1 - h_f s_f)$ \\
$x_4 = ab$ & $(1 - h_m s_m)^2$ & $(1 - s_m)(1 - h_m s_m)$ & $(1-s_m)(1 - h_m s_m)$ & $(1-s_m)^2$ \\
\hline

\end{tabular}
\bigskip{}
\end{table}


\newpage{}

\section*{Figures}
 
\begin{figure}[H] 
\includegraphics[scale=0.8]{./Fig1}
\caption{Self-fertilization increases the scope for genetic linkage to expand the parameter space where balancing selection can maintain SA polymorphism. Results are shown for the conditions of additive allelic effects ($h_i = 1/2$; panels A--C), and dominance reversal ($h_i = 1/4$; panels D--F). In each panel, the shaded region between the black lines indicates the region of sexually antagonistic polymorphism for the single locus case; greyscale lines indicate the thresholds for invasion for female-beneficial (lower lines) and male-beneficial (upper lines) alleles at different recombination rates for the two-locus model.}
\label{fig:funnelPlots}
\end{figure}
\newpage{}


\begin{figure}[H]
\includegraphics[scale=0.9]{./Fig2}
\caption{The proportion of parameter space where SA polymorphism is maintained declines with increased recombination and self-fertilization (panels A,C). The increase in SA polymorphism in the two-locus model relative to the single-locus case also declines with the recombination rate, but increases with the rate of self-fertilization (panels B,D). Results are shown for the conditions of additive allelic effects ($h_i = 1/2$; panels A--B), and dominance reversal ($h_i = 1/4$; panels C--D). Results were obtained by evaluating the leading eigenvalues of $\mathbf{J}$ for populations initially fixed for the haplotypies $[AB]$ and $ab$ at 30,000 points distributed uniformly throughout parameter space defined by $s_m \times s_f$.}
\label{fig:polymorphism}
\end{figure}
\newpage{}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Supporting Information}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Appendix A: Development of recursions, quasi-equilibrium approximations}
\renewcommand{\theequation}{A\arabic{equation}}
\titleformat{\subsubsection}    
{\normalfont\fontsize{12pt}{17}\itshape}{\thesubsubsection}{12pt}{}

\subsubsection*{Recursions}

Consider the genetic system described in the main text. Adult genotypic frequencies formed by the union of the $i$th and $j$th haplotypes are denoted using $F_{ij}$, and the overall haplotype frequencies, $q_i$ are

\begin{align*}
	q_1 &= F_{11} + \frac{F_{12} + F_{13} + F_{14}}{2} \\
	q_2 &= F_{22} + \frac{F_{12} + F_{23} + F_{24}}{2} \\
	q_3 &= F_{33} + \frac{F_{13} + F_{23} + F_{34}}{2} \\
	q_4 &= F_{44} + \frac{F_{14} + F_{24} + F_{34}}{2}. \numberthis \label{eqn}
\end{align*} 

Haplotype frequencies in female gametes are denoted $x_1 = [AB]$, $x_2 = [Ab]$, $x_3 = [aB]$, and $x_4 = [ab]$, and the corresponding frequencies in male gametes are denoted $y_1$, $y_2$, $y_3$, and $y_4$. The contribution of genotype $ij$ to the production of offspring in the next generation is denoted by $F^f_{ij}$. Given these conditions, the recursion equations for the genotypic frequencies in the next generation are (\citealt{Holden1979, JordanConn2014})

\begin{align*} \label{eq:genRec}
	F'_{11} &= (1 - C) (x_1 y_1)           + C \bigg( F^f_{11} + \frac{F^f_{12} + F^f_{13} + F^f_{14}(1 - r)^2 + F^f_{23} r^2}{4} \bigg)  \\
	F'_{12} &= (1 - C) (x_1 y_2 + x_2 y_1) + C \bigg( \frac{F^f_{12} + F^f_{14} r(1-r) + F^f_{23} r(1-r)}{2} \bigg)  \\
	F'_{13} &= (1 - C) (x_1 y_3 + x_3 y_1) + C \bigg( \frac{F^f_{13} + F^f_{14} r(1-r) + F^f_{23} r(1-r)}{2} \bigg)  \\
	F'_{14} &= (1 - C) (x_1 y_4 + x_4 y_1) + C \bigg( \frac{F^f_{14} (1-r)^2 + F^f_{23} r^2}{2} \bigg)  \\
	F'_{22} &= (1 - C) (x_2 y_2)           + C \bigg( F^f_{22} + \frac{F^f_{12} + F^f_{14} r^2 + F^f_{23} (1-r)^2 + F^f_{24}}{4} \bigg)  \\
	F'_{23} &= (1 - C) (x_2 y_3 + x_3 y_2) + C \bigg( \frac{F^f_{14} r^2 + F^f_{23} (1-r)^2}{2} \bigg)  \\
	F'_{24} &= (1 - C) (x_2 y_4 + x_4 y_2) + C \bigg( \frac{F^f_{24} + F^f_{14} r(1-r) + F^f_{23} r(1-r)}{2} \bigg)  \\
	F'_{33} &= (1 - C) (x_3 y_3)           + C \bigg( F^f_{33} + \frac{F^f_{13} + F^f_{14} r^2 + F^f_{23} (1-r)^2 + F^f_{34}}{4} \bigg)  \\
	F'_{34} &= (1 - C) (x_3 y_4 + x_4 y_3) + C \bigg( \frac{F^f_{14} r(1-r) + F^f_{23} r(1-r) + F^f_{34}}{2} \bigg)  \\
	F'_{44} &= (1 - C) (x_4 y_4)           + C \bigg( F^f_{44} + \frac{F^f_{14} (1-r)^2 + F^f_{23} r^2 + F^f_{24} + F^f_{34}}{4} \bigg)  \numberthis
\end{align*}

\noindent{} where $x_{i}$ are

\begin{align*}
	x_1 &= \frac{2 F_{11} w_{f11} + F_{12} w_{f12} + F_{13} w_{f13} + F_{14} w_{f14}}{2 \overline{w}_f} - r \bigg( \frac{F_{14} w_{f14} - F_{23} w_{f14}}{2 \overline{w}_f} \bigg) \\
	x_2 &= \frac{2 F_{22} w_{f22} + F_{12} w_{f12} + F_{23} w_{f23} + F_{24} w_{f24}}{2 \overline{w}_f} + r \bigg( \frac{F_{14} w_{f14} - F_{23} w_{f23}}{2 \overline{w}_f} \bigg) \\
	x_3 &= \frac{2 F_{33} w_{f33} + F_{34} w_{f34} + F_{13} w_{f13} + F_{23} w_{f23}}{2 \overline{w}_f} + r \bigg( \frac{F_{14} w_{f14} - F_{23} w_{f23}}{2 \overline{w}_f} \bigg) \\
	x_4 &= \frac{2 F_{44} w_{f44} + F_{32} w_{f32} + F_{14} w_{f14} + F_{24} w_{f24}}{2 \overline{w}_f} - r \bigg( \frac{F_{14} w_{f14} - F_{23} w_{f23}}{2 \overline{w}_f} \bigg), \numberthis
\end{align*}

\noindent{} $y_{i}$ are

\begin{align*}
	y_1 &= \frac{2 F_{11} w_{m11} + F_{12} w_{m12} + F_{13} w_{m13} + F_{14} w_{m14}}{2 \overline{w}_m} - r \bigg( \frac{F_{14} w_{m14} - F_{23} w_{m14}}{2 \overline{w}_m} \bigg) \\
	y_2 &= \frac{2 F_{22} w_{m22} + F_{12} w_{m12} + F_{23} w_{m23} + F_{24} w_{m24}}{2 \overline{w}_m} + r \bigg( \frac{F_{14} w_{m14} - F_{23} w_{m23}}{2 \overline{w}_m} \bigg) \\
	y_3 &= \frac{2 F_{33} w_{m33} + F_{34} w_{m34} + F_{13} w_{m13} + F_{23} w_{m23}}{2 \overline{w}_m} + r \bigg( \frac{F_{14} w_{m14} - F_{23} w_{m23}}{2 \overline{w}_m} \bigg) \\
	y_4 &= \frac{2 F_{44} w_{m44} + F_{32} w_{m32} + F_{14} w_{m14} + F_{24} w_{m24}}{2 \overline{w}_m} - r \bigg( \frac{F_{14} w_{m14} - F_{23} w_{m23}}{2 \overline{w}_m} \bigg), \numberthis
\end{align*}

\noindent{} where ${w}_{k,ij}$ represent the fitness through each sex function ($k \in [m,f]$) of adults resulting from the union of the $i$th and $j$th haplotypes, and the population mean fitness through each sex function is

\begin{align*}
	\overline{w}_f = F_{11} w_{f11} + &F_{12} w_{f12} + F_{13} w_{f13} + F_{14} w_{f14} + F_{22} w_{f22} + \\ 
				     &F_{23} w_{f23} + F_{24} w_{f24} + F_{33} w_{f33} + F_{34} w_{f34} + F_{44} w_{f44} \numberthis \\
	\overline{w}_m = F_{11} w_{m11} + &F_{12} w_{m12} + F_{13} w_{m13} + F_{14} w_{m14} + F_{22} w_{m22} + \\
				     &F_{23} w_{m23} + F_{24} w_{m24} + F_{33} w_{m33} + F_{34} w_{m34} + F_{44} w_{m44}. \numberthis
\end{align*}


Combining Eqns(A1-A6) and simplifying yields the general haplotype recursion equations: 

\begin{align*} \label{eq:hapRec}
	q'_1 &= F'_{11} + \frac{F'_{12} + F'_{13} + F'_{14}}{2} = (1 - C) \frac{(x_1 + y_1)}{2} + C \bigg( \frac{x_1 - r(F_{14} - F_{23})}{2 \overline{w}_f} \bigg) \\
	q'_2 &= F'_{22} + \frac{F'_{12} + F'_{23} + F'_{24}}{2} = (1 - C) \frac{(x_2 + y_2)}{2} + C \bigg( \frac{x_2 + r(F_{14} - F_{23})}{2 \overline{w}_f} \bigg) \\
	q'_3 &= F'_{33} + \frac{F'_{13} + F'_{23} + F'_{34}}{2} = (1 - C) \frac{(x_3 + y_3)}{2} + C \bigg( \frac{x_3 + r(F_{14} - F_{23})}{2 \overline{w}_f} \bigg) \\
	q'_4 &= F'_{44} + \frac{F'_{14} + F'_{24} + F'_{34}}{2} = (1 - C) \frac{(x_4 + y_4)}{2} + C \bigg( \frac{x_4 - r(F_{14} - F_{23})}{2 \overline{w}_f} \bigg). \numberthis
\end{align*}


\subsubsection*{QE Approximations}

After \citet{CaballeroHill1992}, the QE genotypic frequencies for a single locus, $\mathbf{A}$, are

\begin{align*}
	F_{AA}^* &= q^2 + \frac{C q(1 - q)}{(2 - C)} \\
	F_{Aa}^* &= 2 q (1 - q) - \frac{2 C q(1 - q)}{(2 - C)} \\
	F_{aa}^* &= (1 - q)^2 + \frac{C q(1 - q)}{2 - C}, \numberthis 
\end{align*}

\noindent{} where $q$ is the allele frequency and $C$ is the fixed proportion of self fertilization. We can extend these single-locus results for the two-locus model to calculate the QE adult genotypic frequencies formed by the union of the $i$th and $j$th haplotypes:

\begin{align*}
	\phi_{11} &= q_1^2 + C q_1 (1 - q_1) / (2-C)  \\
	\phi_{12} &= 4 q_1 q_2 (1 - C) / (2-C)        \\
	\phi_{13} &= 4 q_1 q_3 (1 - C) / (2-C)        \\
	\phi_{14} &= 4 q_1 q_4 (1 - C) / (2-C)        \\
	\phi_{22} &= q_2^2 + C q_2 (1 - q_2) / (2-C)  \\
	\phi_{23} &= 4 q_2 q_3 (1 - C) / (2-C)        \\
	\phi_{24} &= 4 q_2 q_4 (1 - C) / (2-C)        \\
	\phi_{33} &= q_3^2 + C q_3 (1 - q_3) / (2-C)  \\
	\phi_{34} &= 4 q_3 q_4 (1 - C) / (2-C)        \\
	\phi_{44} &= q_4^2 + C q_4 (1 - q_4) / (2-C)  \numberthis
\end{align*}

Following standard two-locus theory for partially selfing populations (\citealt{Holden1979, OttoDay2007, JordanConn2014}), and substituting the QE genotypic frequencies, $\phi_{ij}$, yields the haplotype recursions given in eqation (\ref{eq:QEhapRec}), where $x_{i}$ are

\begin{align*}
	x_1 &= \frac{2 \phi_{11} w_{f11} + \phi_{12} w_{f12} + \phi_{13} w_{f13} + \phi_{14} w_{f14}}{2 \overline{w}_f} -
				r \bigg( \frac{\phi_{14} w_{f14} - \phi_{23} w_{f23}} {2 \overline{w}_f} \bigg) \\
	x_2 &= \frac{2 \phi_{22} w_{f22} + \phi_{12} w_{f12} + \phi_{23} w_{f23} + \phi_{24} w_{f24}}{2 \overline{w}_f} +
				r \bigg( \frac{\phi_{14} w_{f14} - \phi_{23} w_{f23}} {2 \overline{w}_f} \bigg) \\ 
	x_3 &= \frac{2 \phi_{33} w_{f33} + \phi_{34} w_{f34} + \phi_{13} w_{f13} + \phi_{23} w_{f23}}{2 \overline{w}_f} +
				r \bigg( \frac{\phi_{14} w_{f14} - \phi_{23} w_{f23}} {2 \overline{w}_f} \bigg) \\
	x_4 &= \frac{2 \phi_{44} w_{f44} + \phi_{34} w_{f34} + \phi_{13} w_{f13} + \phi_{24} w_{f24}}{2 \overline{w}_f} -
				r \bigg( \frac{\phi_{14} w_{f14} - \phi_{23} w_{f23}} {2 \overline{w}_f} \bigg) \numberthis
\end{align*}

\noindent{} and $y_{i}$ are

\begin{align*}
	y_1 &= \frac{2 \phi_{11} w_{m11} + \phi_{12} w_{m12} + \phi_{13} w_{m13} + \phi_{14} w_{m14}}{2 \overline{w}_m} -
				r \bigg( \frac{\phi_{14} w_{m14} - \phi_{23} w_{m23}} {2 \overline{w}_m} \bigg) \\
	y_2 &= \frac{2 \phi_{22} w_{m22} + \phi_{12} w_{m12} + \phi_{23} w_{m23} + \phi_{24} w_{m24}}{2 \overline{w}_m} +
				r \bigg( \frac{\phi_{14} w_{m14} - \phi_{23} w_{m23}} {2 \overline{w}_m} \bigg) \\
	y_3 &= \frac{2 \phi_{33} w_{m33} + \phi_{34} w_{m34} + \phi_{13} w_{m13} + \phi_{23} w_{m23}}{2 \overline{w}_m} +
				r \bigg( \frac{\phi_{14} w_{m14} - \phi_{23} w_{m23}} {2 \overline{w}_m} \bigg) \\
	y_4 &= \frac{2 \phi_{44} w_{m44} + \phi_{34} w_{m34} + \phi_{13} w_{m13} + \phi_{24} w_{m24}}{2 \overline{w}_m} -
				r \bigg( \frac{\phi_{14} w_{m14} - \phi_{23} w_{m23}} {2 \overline{w}_m} \bigg) \numberthis
\end{align*}

\noindent{} and the population mean fitness through each sex function is

\begin{align*}
	\overline{w}_f = \phi_{11} w_{f11} + &\phi_{12} w_{f12} + \phi_{13} w_{f13} + \phi_{14} w_{f14} + \phi_{22} w_{f22} + \\ 
				     &\phi_{23} w_{f23} + \phi_{24} w_{f24} + \phi_{33} w_{f33} + \phi_{34} w_{f34} + \phi_{44} w_{f44} \numberthis \\
	\overline{w}_m = \phi_{11} w_{m11} + &\phi_{12} w_{m12} + \phi_{13} w_{m13} + \phi_{14} w_{m14} + \phi_{22} w_{m22} + \\
				     &\phi_{23} w_{m23} + \phi_{24} w_{m24} + \phi_{33} w_{m33} + \phi_{34} w_{m34} + \phi_{44} w_{m44}. \numberthis
\end{align*}





%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Appendix B: Supplementary figures}
\renewcommand{\theequation}{B\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thefigure}{B\arabic{figure}}
\setcounter{figure}{0}

\begin{figure}[H]
\includegraphics[scale=0.7]{./recSimFig_add}
\caption{Haplotype recursions evaluated at QE, and the resulting invasion conditions, approximate the evolutionary trajectory of the full genotypic recursions very well under additive allelic effects, even under strong selection. Predicted regions of SA polymorphism based on deterministic simulations using the genotypic recursions equations (\ref{eq:genRec}) are compared against the outcome of the invasion analysis for the haplotype recursions using the QE approximation (equations \ref{eq:QEhapRec}) across a gradient of selfing ($C$) and recombination ($r$) rates. Green points indicate parameter conditions where deterministic simulations of the genotypic recursions (Sim.) and the invasion analysis based on eigenvalues for the QE haplotype approximations (Eig.) both predicted polymorphism. Red points indicate regions where the Sim.~predicted polymorphism but the Eig.~did not; and blue points indicate the opposite. The proportion of each outcome is shown in the upper left corner of each panel. Black solid lines show the invasion conditions based on the haplotype recursions using the QE approximation for the given values of $C$ and $r$, with lines drawn for both the two-locus and single-locus invasion conditions when $r > 0$.}
\label{fig:addSim}
\end{figure}
\newpage{}


\begin{figure}[H] 
\includegraphics[scale=0.7]{./recSimFig_domRev}
\caption{Haplotype recursions evaluated at QE, and the resulting invasion conditions, approximate the evolutionary trajectory of the full genotypic recursions very well under allelic effects resulting in dominance reversal, even under strong selection. Predicted regions of SA polymorphism based on deterministic simulations using the genotypic recursions equations (\ref{eq:genRec}) are compared against the outcome of the invasion analysis for the haplotype recursions using the QE approximation (equations \ref{eq:QEhapRec}) across a gradient of selfing ($C$) and recombination ($r$) rates. Green points indicate parameter conditions where deterministic simulations of the genotypic recursions (Sim.) and the invasion analysis based on eigenvalues for the QE haplotype approximations (Eig.) both predicted polymorphism. Red points indicate regions where the Sim.~predicted polymorphism but the Eig.~did not; and blue points indicate the opposite. The proportion of each outcome is shown in the upper left corner of each panel. Black solid lines show the invasion conditions based on the haplotype recursions using the QE approximation for the given values of $C$ and $r$, with lines drawn for both the two-locus and single-locus invasion conditions when $r > 0$.}
\label{fig:domRevSim}
\end{figure}
\newpage{}


\begin{figure}[H]
\includegraphics[scale=0.75]{./Fig1wk}
\caption{Under weak selection ($s_f,s_m < 0.1$), the parameter space where sexually antagonistic polymorphism is maintained is only expanded beyond the single-locus case when there is very tight linkage ($r \approx 0$). Thus, there is little scope for self-fertilization to influence sexually antagonistic polymorphism. Results are shown for the conditions of additive allelic effects ($h_i = 1/2$; panels A--C), and dominance reversal ($h_i = 1/4$; panels D--F). In each panel, the shaded region between the black lines indicates the region of sexually antagonistic polymorphism for the single-locus case; grey lines indicate the thresholds for invasion for female-beneficial (lower lines) and male-beneficial (upper lines) alleles under complete linkage ($r = 0$).}
\label{fig:wkFunnelPlots}
\end{figure}
\newpage{}


\begin{figure}[H]
\includegraphics[scale=0.8]{./Fig2wk}
\caption{Under weak selection, the proportion of parameter space where SA polymorphism is maintained again declines with increased recombination and self-fertilization (panels A,C). The increase in SA polymorphism in the two-locus model relative to the single-locus case also declines with the recombination rate, but increases with the rate of self-fertilization (panels B,D). However, these effects are limited to cases of extremely tight linkage ($r < 0.004$). Results are shown for the conditions of additive allelic effects ($h_i = 1/2$; panels A--B), and dominance reversal ($h_i = 1/4$; panels C--D). Results were obtained by evaluating the leading eigenvalues of $\mathbf{J}$ for populations initially fixed for the haplotypies $[AB]$ and $ab$ at 30,000 points distributed uniformly throughout parameter space defined by $s_m \times s_f$, where $s_f,s_m \in [0,0.1]$}
\label{fig:wkPolymorhism}
\end{figure}
\newpage{}

%%%%%%%%%%%%%%%%%%%%%
% Bibliography
%%%%%%%%%%%%%%%%%%%%%

\begin{thebibliography}{}

\bibitem[{Abbott(2011)Abbott}]{Abbott2011}
Abbott, J.~K. 2011.
\newblock Intralocus sexual conflict and sexually antagonistic genetic variation in hermaphroditic animals.
\newblock Proc. R. Soc. B 278:161–-169.

\bibitem[{Agrawal and Whitlock(2011)Agrawal and Whitlock}]{AgrawalWhitlock2011}
Agrawal, A.~F., and M.~C. Whitlock. 2011.
\newblock Inferences about the distribution of dominance drawn from yeast gene knockout data.
\newblock Genetics 178:553--566.

\bibitem[{Barrett(2002)Barrett}]{Barrett2002}
Barrett, S.~C.~H. 2002.
\newblock Sexual interference of the floral kind.
\newblock Heredity 88:154–-159.

\bibitem[{Barson et~al.(2015)Barson et al. 2015}]{Barson2015}
Barson, N.~J. \textit{et~al}. 2015.
\newblock Sex-dependent dominance at a single locus maintains variation in age at maturity in Atlantic salmon.
\newblock Nature 528:405--408.

\bibitem[{Bonduriansky and Chenoweth(2009)Bonduriansky and Chenoweth}]{BondChen2009}
Bonduriansky, R. and S.~F. Chenoweth. 2009.
\newblock intralocus sexual conflict.
\newblock Trends in Ecology and Evolution 24:280--288.

\bibitem[{Caballero and Hill(1992)Caballero and Hill}]{CaballeroHill1992}
Caballero, A. and W.~G. Hill. 1992.
\newblock Effects of partial inbreeding on fixation rates and variation of mutant genes.
\newblock Genetics 131:493--507.

\bibitem[{Charlesworth and Charlesworth(1978)Charlesworth and Charlesworth}]{Charlesworth1978}
Charlesworth, D. and B. Charlesworth. 1978.
\newblock Population genetics of partial male-sterility and the evolution of monoecy and dioecy.
\newblock Heredity 41:137--153.

\bibitem[{Charlesworth and Hughes(1999)Charlesworth and Hughes}]{CharlesworthHughes1999}
Charlesworth, B. and K.~A. Hughes. 1999.
\newblock The maintenance of genetic variation in life-history traits.
\newblock Pp. 369-–392 in R.~S. Singh and C.~B. Krimbas, \textit{eds}. Evolutionary genetics: from molecules to morphology. Vol. 1. Cambridge Univ. Press, Cambridge, U.K.

\bibitem[{Connallon and Clark(2011)Connallon and Clark}]{ConnClark2011}
Connallon, T. and A.~G. Clark. 2011.
\newblock The resolution of sexual antagonism by gene duplication
\newblock Genetics 187:919--937.

\bibitem[{Connallon and Clark(2012)Connallon and Clark}]{ConnClark2012}
Connallon, T. and A.~G. Clark. 2012.
\newblock A general population genetic framework for antagonistic selection that accounts for demography and recurrent mutation.
\newblock Genetics 190:1477--1489.

\bibitem[{Connallon and Clark(2014)Connallon and Clark}]{ConnClark2014}
Connallon, T. and A.~G. Clark. 2014b.
\newblock Balancing selection in species with separate sexes: insights from Fisher's geometric model.
\newblock Genetics 197:991--1006.

\bibitem[{Conner(2006)Conner}]{Conner2006}
Conner, J. 2006.
\newblock Ecological genetics of floral evolution.
\newblock Pp. 260-–277 in L.~D. Harder and S.~C.~H. Barrett, \textit{eds}. The ecology and evolution of flowers. Oxford Univ. Press, New York.

\bibitem[{Cox and Calsbeek(2010)Cox and Calsbeek}]{CoxCals2010}
Cox, R.~M. and R. Calsbeek. 2010.
\newblock Cryptic sex-ratio bias provides indirect genetic benefits despite sexual conflict.
\newblock Science 328:92--94.

\bibitem[{Fry(2010)Fry}]{Fry2010}
Fry, J.~D. 2010.
\newblock The genomic location of sexually antagonistic variation: some cautionary comments.
\newblock Evolution 64:1510-1516.

\bibitem[{Goodwillie et al.(2005)Goodwillie et al.}]{Goodwillie2005}
Goodwillie, C., S. Kalisz, and C.~G. Eckert. 2005.
\newblock The evolutionary enigma of mixed mating systems in plants: occurrence, theoretical explanations, and empirical evidence.
\newblock Ann. Rev. Ecol. Evol. Syst. 36:47--79.

\bibitem[{Harder and Barrett(2006)Harder and Barrett}]{HarderBarrett2006} %Need to choose chapter!
Harder, L.~D. and S.~C.~H. Barrett. 2006.
\newblock Ecology and evolution of flowers.
\newblock Pp. 369-–392 in R.~S. Singh and C.~B. Krimbas, \textit{eds}. Evolutionary genetics: from molecules to morphology. Vol. 1. Cambridge Univ. Press, Cambridge, U.K.

\bibitem[{Holden(1979)Holden}]{Holden1979}
Holden, L.~R. 1979.
\newblock New properties of the two-locus partial selfing model with selection.
\newblock Genetics 93:217--236.

\bibitem[{Igic and Kohn(2005)Igic and Kohn}]{Igic2005}
Igic, B. and J.~R. Kohn. 2005.
\newblock The distribution of plant mating systems: study bias against obligately outcrossing species.
\newblock Evolution 60:1098--1103.

\bibitem[{Jarne and Auld(2006)Jarne and Auld}]{JarneAuld2006}
Jarne, P. and J.~R. Auld. 2006.
\newblock Animals mix it up too: the distribution of self-fertilization among hermaphroditic animals.
\newblock Evolution 60:1816--1824.

\bibitem[{Jordan and Charlesworth(2012)Jordan and Charlesworth}]{JordanCharles2012}
Jordan, C.~Y. and D. Charlesworth. 2012.
\newblock The potential for sexually antagonistic polymorphism in different genome regions.
\newblock Evolution 66:505--516.

\bibitem[{Jordan and Connallon(2014)Jordan and Connallon}]{JordanConn2014}
Jordan, C.~Y. and T. Connallon. 2014.
\newblock Sexually antagonistic polymorphism in simultaneous hermaphrod-\\ites.
\newblock Evolution 68:3555--3569.

\bibitem[{Kidwell et~al.(1977)Kidwell, Clegg, Stewart, and Prout}]{Kidwell1977}
Kidwell, J.~F., M.~T. Clegg, F.~M. Stewart, and T. Prout. 1977.
\newblock Regions of stable equilibria for models of differential selection in the two sexes under random mating.
\newblock Genetics 85:171--183.

\bibitem[{Kimura and Ohta(1971)Kimura and Ohta}]{KimuraOhta1971}
Kimura, M. and T. Ohta. 1971.
\newblock Theoretical aspects of population genetics.
\newblock Princeton University Press, Princeton, New Jersey, USA.

\bibitem[{Lloyd and Webb(1986)Lloyd and Webb}]{LloydWebb1986}
D.~G. Lloyd and Webb, C.~J. 1986.
\newblock The avoidance of interference between the presentation of pollen and stigmas in angiosperms I. Dichogamy.
\newblock NZ J. Bot. 24:135--162.

\bibitem[{Manna et.~al.(2011)Manna et.~al.}]{Manna2011}
F. Manna, G. Martin, and T. Lenormand. 2011.
\newblock Fitness landscapes: an alternative theory for the dominance of mutation.
\newblock Genetics 189:923--937.

\bibitem[{Nagylaki(1997)Nagylaki}]{Nagylaki1997}
Nagylaki, T. 1997.
\newblock The diffusion model for migration and selection in a plant population.
\newblock Journal of Mathematical Biology 35:409--431.

\bibitem[{Otto and Day(2007)Otto and Day}]{OttoDay2007}
Otto, S.~P. and T. Day. 2007.
\newblock A biologist's guide to mathematical modeling in ecology and evolution.
\newblock Princeton University Press, Princeton, New Jersey, USA.

\bibitem[{Patten et~al.(2010)Patten et~al.}]{Patten2010}
M.~M. Patten, D. Haig, and F. \'{U}beda. 2010.
\newblock Fitness variation due to sexual antagonism and linkage disequilibrium.
\newblock Evolution 64:3638--3642.

\bibitem[{Prout(1968)Prout}]{Prout1968}
Prout, T. 1968.
\newblock Sufficient conditions for multiple niche polymorphism.
\newblock American Naturalist 102:493--496.

\bibitem[{Prout(2000)Prout}]{Prout2000}
Prout, T. 2000.
\newblock How well does opposing selection maintain variation?
\newblock Pp. 157--203 \textit{in} R.~S. Singh and C.~B. Kimbras, eds. Evolutionary Genetics from molecules to morphology. Cambridge University Press, Cambridge, U.K.

\bibitem[{Rice(1992)Rice}]{Rice1992}
Rice, W.~R. 1992.
\newblock Sexually antagonistic genes: experimental evidence.
\newblock Science 256:1436--1439.

\bibitem[{Rice and Chippindale(2001)Rice and Chippindale}]{RiceChipp2001}
Rice, W.~R. and A.~K. Chippindale. 2001.
\newblock Intersexual ontogenetic conflict.
\newblock Journal of evolutionary biology 14:685--693.

\bibitem[{Sicard and Lenhard(2011)Sicard and Lenhard}]{Sicard2011}
Sicard, A. and M. Lenhard. 2011.
\newblock The selfing syndrome: a model for studying the genetic and evolutionary basis of morphological adaptation in plants.
\newblock Ann. Bot. 107:1433--1443.

\bibitem[{Spencer and Priest(2016)Spencer and Priest}]{Spencer2016}
Spencer, H.~G. and N.~K. Priest. 2016.
\newblock The evolution of sex-specific dominance in response to sexually antagonistic selection.
\newblock The American Naturalist 187:658--666.

\bibitem[{Tazzyman and Abbott(2015)Tazzyman and Abbott}]{Tazzyman2015}
Tazzyman, S.~J. and J.~K. Abbott. 2015.
\newblock Self-fertilization and inbreeding limit the scope for sexually antagonisitc polymorphism.
\newblock Journal of Evolutionary Biology 28:723--729.

\bibitem[{\'Ubeda et.~al.(2010)\'Ubeda et.~al.}]{Ubeda2010}
\'Ubeda, F., D. Haig, and M.~M. Patten. 2010.
\newblock Stable linkage disequilibrium owing to sexual antagonism.
\newblock Proc.~Roy.~Soc.~B DOI:10.1098/rspb.2010.12.

\bibitem[{Webb and Lloyd(1986)Webb and Lloyd}]{WebbLloyd1986}
Webb, C.~J. and D.~G. Lloyd. 1986.
\newblock The avoidance of interference between the presentation of pollen and stigmas in angiosperms II. Herkogamy.
\newblock NZ J. Bot. 24:163--178.

\bibitem[{Wright et.~al.(2008)Wright et.~al.}]{Wright2008}
Wright, S.~I., R.~W. Ness, J.~P. Foxe, and S.~C.~H. Barrett. 2008.
\newblock Genomic consequences of outcrossing and selfing in plants.
\newblock Int. J. Plant Sci. 169:105--118.



\end{thebibliography}

\end{document}
